### DISCUSSION TITLE
HTTP status code (500) on add site

### DISCUSSION THREAD

**Discussion by komir2004** [akeeba/panopticon#370] [Q&A (Community Support)] [answered]
Hi, I'm having a problem with Panopticon. I'm using Joomla 4.4.1 and I want to add a web page to it, but I get an error message. 
```
The Joomla! API Application responded with an unexpected HTTP status code (500)
Most likely a system or webservices plugin on your server modified the request we sent, resulting in an error. Please check your installed plugins.
Before you attempt to submit an issue about this connection issue please go to Administration, System Configuration, System and enable Debug System. Then come back here and retry the same connection. You will receive detailed troubleshooting information which you need to include in your issue.
```
Endpoint URL is start with https://www. as my page. I tried to enable debug mode to see what is causing the error, but nothing changed on the site. I also added the api code in the .htaccess Maker section of Admin Tools, but it didn't help. The Admin Tools firewall does not show any blocked requests. Webservices plugin is enabled.

I contacted my hosting company, but they said they don't see anything wrong in their Firewall or mod_security settings. 
Does anyone know how to fix this or some other steps to debug it?

**Comment by nikosdion** [comment]
As per the message, one of your site's plugins is causing the problem. Setting Error Reporting to Development and Debug Site to Yes might give you some clue, or at least output something in your server's error logs. In the latter case, ask your host to look at the server's error logs for a debug trace when that error 500 is triggered.

**Comment by komir2004** [comment]
I appreciate your help, but I'm still stuck with the same error.
I tried changing the Setting Error Reporting to Development and Debug Site to Yes, but it didn't make any difference.
My hosting company told me there are 4 mod_security rules that might be causing the problem, but even after they whitelisted them, nothing changed.
On the targeted web page there is no Admintools installed.
Is there any other way I can debug this issue?

**Comment by nikosdion** [comment]
Have you tried accessing the API URL directly? Do you get an error?

**Comment by komir2004** [comment]
Get

`{
    "errors": [
        {
            "title": "Resource not found",
            "code": 404
        }
    ]
}`

**Comment by nikosdion** [comment]
Darn it. This means that you only get a problem with authenticated API calls. Can you use the contact form on either my business or my blog to contact me with the URL to your site and the API key? **Do not post them on GitHub**. Unfortunately, there's no way to send private messages on GitHub :(

**Comment by nikosdion** [comment, marked as answer]
A short update for everyone else reading this discussion.

After being given connection details to the two problem sites we discovered the following:

Site 1: EasyBlog's user plugin breaks the Joomla! API application because it idiotically assumes that it's the same as the site application (site frontend). However, the API application **does not** have a menu, therefore when the plugin tries to access the non-existent menu it crashes Joomla!. For those of you that have been following my writings, [it is exactly what I warned about on June 27h, 2022](https://www.dionysopoulos.me/common-mistakes-writing-joomla-plugins.html) and had been talking about for well over two years prior. Ahem.

Site 2: The site's files were all messed up. Trying to access the `/api` URL on the site resulted in a Joomla! 404 Not found error page. That should have been a strong indication that the site has soiled itself, but I digress. I used Joomla Update's feature to reinstall the core files, and @komir2004 regenerated the .htaccess file. The site connected just fine.

Please keep in mind that connection issues are always a problem on _your site_. Anything else is vanishingly rare. When you have a problem, always check your site. Panopticon 1.0.6 will have a Connection Doctor page which will show you the exact output it receives from your site so you can do self-troubleshooting (that was what was necessary for Site 1 in this case). 

Please exhaust all of your troubleshooting mojo before asking the Panopticon community for help. Community members are unlikely to be able or want to help you, and I am just one person who's already wearing too many hats.
